Documentation/coding-guidelines/interop-pinvokes.md

```diff
@@ -218,6 +225,64 @@ Blittable structs are much more performant as they they can simply be used direc
 
 Pointers to structs in definitions must either be passed by `ref` or use `unsafe` and `*`.
 
+The compiler or marshaler will make sure any pointers in your struct are correctly aligned. For this reason seemingly "wrong" declarations can still work. For example PROCESS_BASIC_INFORMATION is documented as:
+
+```C
+typedef struct _PROCESS_BASIC_INFORMATION {
+    PVOID Reserved1;
+    PPEB PebBaseAddress;
+    PVOID Reserved2[2];
+    ULONG_PTR UniqueProcessId;
+    PVOID Reserved3;
+} PROCESS_BASIC_INFORMATION;
+```
+
+All the members of this struct have the underlying type of pointer, so a correct managed representation could be:
+
+```c#
+    struct PROCESS_BASIC_INFORMATION
+    {
+        public IntPtr Reserved1; // or UIntPtr
+        public IntPtr PebBaseAddress;
+        public IntPtr Reserved2a;
+        public IntPtr Reserved2b;
+        public IntPtr BasePriority;
+        public UIntPtr UniqueProcessId;
+        public UIntPtr Reserved3;
+    }
+```
+
+Under the covers, the first field in this struct is actually type `LONG` which would normally marshal as a 32-bit `int` in managed code. The reason that `IntPtr` still works is that the next field happens to be a pointer, which must be aligned. To
+achieve that alignment, the compiler or marshaler would pad the preceding field to pointer width even if it is not declared as a pointer.
+
+In such cases and others, we always prefer to match the managed struct as closely as possible to the shape and names that is used in the documentation. 
+
+In the case above, the array of pointers `PVOID Reserved2[2]` was marshaled to two `IntPtr` fields, `Reserved2a` and `Reserved2b`. When the native array is a primitive type, we can use the `fixed` keyword to write it a little more cleanly. For example, `SYSTEM_PROCESS_INFORMATION` looks like this in the native header:
+
+```c
+typedef struct _SYSTEM_PROCESS_INFORMATION {
+    ULONG NextEntryOffset;
+    ULONG NumberOfThreads;
+    BYTE Reserved1[48];
+    UNICODE_STRING ImageName;
+...
+} SYSTEM_PROCESS_INFORMATION
+```
+
+In C#, we can write it like this:
+
+```c#
+    internal unsafe struct SystemProcessInformation
+    {
+        internal uint NextEntryOffset;
+        internal uint NumberOfThreads;
+        private fixed byte Reserved1[48];
+        internal Interop.UNICODE_STRING ImageName;
+        ...
+    }
+```
+It makes sense to make the undocumented members `private`.
```

Comment:
I do not think this really matters. But if you would like to include it here, make the other examples in this doc (like `PROCESS_BASIC_INFORMATION`) consistent with it.

Reply:
It helps prevent using undocumented members without first thinking about the potential consequences.

Reply:
The undocumented members are sometimes required to be set to 0.

C/C++ does not make the private. If it is fine not make them private for C/C++, it is fine for C# too.